src/NzbDrone.Core/Indexers/HttpIndexerBase.cs

```diff
@@ -485,12 +496,22 @@ protected virtual async Task<ValidationFailure> TestConnection()
                     _logger.Warn(ex, "Indexer does not support the query");
                     return new ValidationFailure(string.Empty, "Indexer does not support the current query. Check if the categories and or searching for movies are supported. Check the log for more details.");
                 }
-                else
-                {
-                    _logger.Warn(ex, "Unable to connect to indexer");
 
-                    return new ValidationFailure(string.Empty, "Unable to connect to indexer, check the log for more details");
-                }
+                _logger.Warn(ex, "Unable to connect to indexer");
+
+                return new ValidationFailure(string.Empty, "Unable to connect to indexer, check the log for more details");
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.Warn(ex, "Unable to connect to indexer");
+
+                return new ValidationFailure(string.Empty, "Unable to connect to indexer, please check your DNS settings. " + ex.Message);
+            }
+            catch (TaskCanceledException ex)
+            {
+                _logger.Warn(ex, "Unable to connect to indexer");
+
+                return new ValidationFailure(string.Empty, "Unable to connect to indexer, possibly due to a timeout. If needed try again. " + ex.Message);
```

Comment:
```suggestion
                return new ValidationFailure(string.Empty, "Unable to connect to indexer due to timeout and no response after 100 seconds. Try again or check your network settings. " + ex.Message);
```

Reply:
Agree with bogdan, let's leave it generic